### PR TITLE
[plugin.video.piped] 1.0.0

### PR DESCRIPTION
**Pull request by syhlx**
### Description
New addon which allows users to access any Piped instance, login and manage their playlists and watch history.

More information about Piped:
[Piped GitHub Repository](https://github.com/TeamPiped/Piped)
[Piped Official Instance](https://piped.video/)

### Checklist:
- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

**Comment by basrieter** [comment]
Please fix the issues pointed out by the add-on checker. Then I will have a further look.

**Comment by syhlx** [comment]
This should fix all the issues pointed out by the add-on checker

File: plugin.video.piped/lib/services/entrypoint.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+from lib.services.player import Player
+
+class Service(xbmc.Monitor):
+	def __init__(self):
+		httpservice = Thread(target=HttpService)
+		scheduler = Scheduler()
+		httpservice.daemon = True
+		scheduler.daemon = True
+		player = Player()
+		player.isPlaying()
```
Comment (basrieter) [inline plugin.video.piped/lib/services/entrypoint.py]: Why this call? Makes no real sense if you don't use the output?

File: plugin.video.piped/lib/services/entrypoint.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+		while not self.abortRequested():
```
Comment (basrieter) [inline plugin.video.piped/lib/services/entrypoint.py]: If `abortRequested()` results in `True`, the `httpservice` and `scheduler` aren't stopped properly and could cause Kodi to wait for a long time.

File: plugin.video.piped/lib/services/entrypoint.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+			if self.waitForAbort(5):
+				while scheduler.is_alive():
+					scheduler.stop()
+					xbmc.sleep(200)
```
Comment (basrieter) [inline plugin.video.piped/lib/services/entrypoint.py]: If `scheduler.stop()` does not finish, the script waits for 200ms and then stops it again? In the `Scheduler.stop()` you call `self.join()` which should actually block until the thread stopped.

**Review by basrieter — CHANGES_REQUESTED** [review]
Could you have a look at the code for the services? It does not seem to handle aborts correctly.

**Comment by syhlx** [comment]
Before I commit anything, does this seem right to you?

```python
class Service(xbmc.Monitor):
  def __init__(self):
    httpservice = Thread(target=HttpService)
    scheduler = Scheduler()
    httpservice.daemon = True
    scheduler.daemon = True
    player = Player()

    while not self.waitForAbort(5):
      try:
        if not httpservice.is_alive(): httpservice.start()
        if not scheduler.is_alive(): scheduler.start()
      except:
        pass

    scheduler.stop()
```

**Comment by basrieter** [comment]
Yes that makes more sense. Also make sure the `httpservice` is stopped in the line under the `scheduler.stop()`

**Comment by syhlx** [comment]
Done

**Review by basrieter — APPROVED** [review]

**Pull request merged**